File: schemaorg-rs/src/health_plan_formulary.rs
use crate::*;
use serde_with::{serde_as, OneOrMany};
///<https://schema.org/additionalType>
///<https://schema.org/Text>
///<https://schema.org/URL>
pub type HealthPlanFormularyAdditionalTypeFieldEnum = String;
///<https://schema.org/HealthPlanFormulary>
#[serde_as]
#[derive(Debug, serde::Deserialize)]
#[cfg_attr(feature = "uniffi", derive(uniffi::Record))]
pub struct HealthPlanFormulary {
    #[serde(rename = "@context")]
    pub context: String,
    ///<https://schema.org/offersPrescriptionByMail>
    #[serde(rename = "offersPrescriptionByMail")]
    #[serde_as(as = "OneOrMany<_>")]
    pub offers_prescription_by_mail: Vec<String>,
    ///<https://schema.org/healthPlanCostSharing>
    #[serde(rename = "healthPlanCostSharing")]
    #[serde_as(as = "OneOrMany<_>")]
    pub health_plan_cost_sharing: Vec<String>,
    ///<https://schema.org/healthPlanDrugTier>
    #[serde(rename = "healthPlanDrugTier")]
    #[serde_as(as = "OneOrMany<_>")]
    pub health_plan_drug_tier: Vec<String>,
    ///<https://schema.org/name>
    #[serde(rename = "name")]
    #[serde_as(as = "OneOrMany<_>")]
    pub name: Vec<String>,
    ///<https://schema.org/mainEntityOfPage>
    #[serde(rename = "mainEntityOfPage")]
    #[serde_as(as = "OneOrMany<_>")]
    pub main_entity_of_page: Vec<HealthPlanFormularyMainEntityOfPageFieldEnum>,
    ///<https://schema.org/url>
    #[serde(rename = "url")]
    #[serde_as(as = "OneOrMany<_>")]
    pub url: Vec<String>,
    ///<https://schema.org/disambiguatingDescription>
    #[serde(rename = "disambiguatingDescription")]
    #[serde_as(as = "OneOrMany<_>")]
    pub disambiguating_description: Vec<String>,
    ///<https://schema.org/identifier>
    #[serde(rename = "identifier")]
    #[serde_as(as = "OneOrMany<_>")]
    pub identifier: Vec<HealthPlanFormularyIdentifierFieldEnum>,
    ///<https://schema.org/description>
    #[serde(rename = "description")]
    #[serde_as(as = "OneOrMany<_>")]
    pub description: Vec<HealthPlanFormularyDescriptionFieldEnum>,
    ///<https://schema.org/subjectOf>
    #[serde(rename = "subjectOf")]
    #[serde_as(as = "OneOrMany<_>")]
    pub subject_of: Vec<HealthPlanFormularySubjectOfFieldEnum>,
    ///<https://schema.org/alternateName>
    #[serde(rename = "alternateName")]
    #[serde_as(as = "OneOrMany<_>")]
    pub alternate_name: Vec<String>,
    ///<https://schema.org/additionalType>
    #[serde(rename = "additionalType")]
    #[serde_as(as = "OneOrMany<_>")]
    pub additional_type: Vec<HealthPlanFormularyAdditionalTypeFieldEnum>,
    ///<https://schema.org/potentialAction>
    #[serde(rename = "potentialAction")]
    #[serde_as(as = "OneOrMany<_>")]
    pub potential_action: Vec<Action>,
    ///<https://schema.org/sameAs>
    #[serde(rename = "sameAs")]
    #[serde_as(as = "OneOrMany<_>")]
    pub same_as: Vec<String>,
    ///<https://schema.org/image>
    #[serde(rename = "image")]
    #[serde_as(as = "OneOrMany<_>")]
    pub image: Vec<HealthPlanFormularyImageFieldEnum>,
}

File: schemaorg-ts/src/health_plan_formulary.ts
import type * as all from "./index";

/**
 * <https://schema.org/additionalType>
 **/
export type HealthPlanFormularyAdditionalTypeFieldEnum = string;

/**
 * <https://schema.org/HealthPlanFormulary>
 **/
export interface HealthPlanFormulary {
  "@context": string;
  "@type": "HealthPlanFormulary";

  /**
   * <https://schema.org/offersPrescriptionByMail>
   **/
  offersPrescriptionByMail: all.OneOrMany<string>;

  /**
   * <https://schema.org/healthPlanCostSharing>
   **/
  healthPlanCostSharing: all.OneOrMany<string>;

  /**
   * <https://schema.org/healthPlanDrugTier>
   **/
  healthPlanDrugTier: all.OneOrMany<string>;

  /**
   * <https://schema.org/name>
   **/
  name: all.OneOrMany<string>;

  /**
   * <https://schema.org/mainEntityOfPage>
   **/
  mainEntityOfPage: all.OneOrMany<all.HealthPlanFormularyMainEntityOfPageFieldEnum>;

  /**
   * <https://schema.org/url>
   **/
  url: all.OneOrMany<string>;

  /**
   * <https://schema.org/disambiguatingDescription>
   **/
  disambiguatingDescription: all.OneOrMany<string>;

  /**
   * <https://schema.org/identifier>
   **/
  identifier: all.OneOrMany<all.HealthPlanFormularyIdentifierFieldEnum>;

  /**
   * <https://schema.org/description>
   **/
  description: all.OneOrMany<all.HealthPlanFormularyDescriptionFieldEnum>;

  /**
   * <https://schema.org/subjectOf>
   **/
  subjectOf: all.OneOrMany<all.HealthPlanFormularySubjectOfFieldEnum>;

  /**
   * <https://schema.org/alternateName>
   **/
  alternateName: all.OneOrMany<string>;

  /**
   * <https://schema.org/additionalType>
   **/
  additionalType: all.OneOrMany<all.HealthPlanFormularyAdditionalTypeFieldEnum>;

  /**
   * <https://schema.org/potentialAction>
   **/
  potentialAction: all.OneOrMany<all.Action>;

  /**
   * <https://schema.org/sameAs>
   **/
  sameAs: all.OneOrMany<string>;

  /**
   * <https://schema.org/image>
   **/
  image: all.OneOrMany<all.HealthPlanFormularyImageFieldEnum>;
}
